pkg/registry/generic/etcd/etcd.go

```diff
@@ -455,3 +445,16 @@ func (e *Etcd) WatchPredicate(ctx api.Context, m generic.Matcher, resourceVersio
 
 	return e.Helper.WatchList(e.KeyRootFunc(ctx), version, filterFunc)
 }
+
+// calculateTTL is a helper for retrieving the updated TTL for an object or returning an error
+// if the TTL cannot be calculated. The defaultTTL is changed to zero if less than zero.
+func (e *Etcd) calculateTTL(obj runtime.Object, defaultTTL int64, update bool) (ttl uint64, err error) {
+	if defaultTTL < 0 {
+		defaultTTL = 0
+	}
+	ttl = uint64(defaultTTL)
+	if e.TTLFunc != nil {
+		ttl, err = e.TTLFunc(obj, ttl, update)
+	}
```

Comment:
Yeah I need to look at the etcd source code but it's either because in rare cases observed TTL can be negative or part of the original spec.  Will add a comment.

> On May 23, 2015, at 1:26 AM, Daniel Smith <notifications@github.com> wrote:
> 
> In pkg/registry/generic/etcd/etcd.go:
> 
> > @@ -455,3 +445,16 @@ func (e *Etcd) WatchPredicate(ctx api.Context, m generic.Matcher, resourceVersio
> >  
> >  	return e.Helper.WatchList(e.KeyRootFunc(ctx), version, filterFunc)
> >  }
> > +
> > +// calculateTTL is a helper for retrieving the updated TTL for an object or returning an error
> > +// if the TTL cannot be calculated. The defaultTTL is changed to zero if less than zero.
> > +func (e *Etcd) calculateTTL(obj runtime.Object, defaultTTL int64, update bool) (ttl uint64, err error) {
> > +	if defaultTTL < 0 {
> > +		defaultTTL = 0
> > +	}
> > +	ttl = uint64(defaultTTL)
> > +	if e.TTLFunc != nil {
> > +		ttl, err = e.TTLFunc(obj, ttl, update)
> > +	}
> So returned ttl is allowed to be > 2^63 - 1, but not the default? comment on this maybe?
> 
> —
> Reply to this email directly or view it on GitHub.
>